app/services/globus_digest_generator.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# GlobusDigestGenerator
+#
+# Add digests to Globus DROs if the files currently lack them. Existing files
+# will not be overwritten.
+class GlobusDigestGenerator
+  # @param [Cocina::Model] cocina a DRO to add digests to
+  # @param [Hash] globus_ids a mapping of filenames to their location on disk
+  # @param [Boolean] validate whether to build the new DRO using validation
+  # @return [Cocina::Model] a new Cocina object with (potentially) new digests
+  def self.generate(cocina:, globus_ids:, validate: true)
+    new(cocina: cocina, globus_ids: globus_ids, validate: validate).generate
+  end
+
+  def initialize(cocina:, globus_ids:, validate:)
+    @cocina = cocina
+    @globus_ids = globus_ids
+    @validate = validate
+  end
+
+  def generate
+    return cocina unless cocina.dro?
+
+    props = cocina.to_h
```

Comment:
I tried using `new` here instead but because it needs to work with a Dro used for creating or updating a Dro I ran into some trouble with it validating or not. So instead I pressed the easy button and converted the whole thing to a Hash. If there is a better way to do this let me know!